src/webgl/p5.RendererGL.js

```diff
@@ -1899,44 +2011,100 @@ p5.RendererGL = class RendererGL extends p5.Renderer {
     *  maps a p5.Image used by imageLight() to a p5.Framebuffer
    */
   getDiffusedTexture(input) {
-    // if one already exists for a given input image
+    // If one already exists for a given input image
     if (this.diffusedTextures.get(input) != null) {
       return this.diffusedTextures.get(input);
     }
-    // if not, only then create one
-    let newFramebuffer;
-    // hardcoded to 200px, because it's going to be blurry and smooth
-    let smallWidth = 200;
-    let width = smallWidth;
-    let height = Math.floor(smallWidth * (input.height / input.width));
-    newFramebuffer = this._pInst.createFramebuffer({
-      width, height, density: 1
-    });
-    // create framebuffer is like making a new sketch, all functions on main
-    // sketch it would be available on framebuffer
-    if (!this.diffusedShader) {
-      this.diffusedShader = this._pInst.createShader(
-        defaultShaders.imageLightVert,
-        defaultShaders.imageLightDiffusedFrag
+
+    // Create a cubemap texture
+    const envCubemap = this.GL.createTexture();
+    this.GL.bindTexture(this.GL.TEXTURE_CUBE_MAP, envCubemap);
+
+    // Loop through each face and allocate storage
+    for (let i = 0; i < 6; ++i) {
+      this.GL.texImage2D(
+        this.GL.TEXTURE_CUBE_MAP_POSITIVE_X + i, 0,
+        this.GL.RGB16F, 512, 512, 0,
+        this.GL.RGB, this.GL.FLOAT, null
       );
     }
-    newFramebuffer.draw(() => {
-      this._pInst.shader(this.diffusedShader);
-      this.diffusedShader.setUniform('environmentMap', input);
-      this._pInst.noStroke();
-      this._pInst.rectMode(constants.CENTER);
-      this._pInst.noLights();
-      this._pInst.rect(0, 0, width, height);
+
+    // Set parameters for the cubemap
+    this.GL.texParameteri(this.GL.TEXTURE_CUBE_MAP
+      , this.GL.TEXTURE_WRAP_S, this.GL.CLAMP_TO_EDGE);
+    this.GL.texParameteri(this.GL.TEXTURE_CUBE_MAP
+      , this.GL.TEXTURE_WRAP_T, this.GL.CLAMP_TO_EDGE);
+    this.GL.texParameteri(this.GL.TEXTURE_CUBE_MAP
+      , this.GL.TEXTURE_WRAP_R, this.GL.CLAMP_TO_EDGE);
+    this.GL.texParameteri(this.GL.TEXTURE_CUBE_MAP
+      , this.GL.TEXTURE_MIN_FILTER, this.GL.LINEAR);
+    this.GL.texParameteri(this.GL.TEXTURE_CUBE_MAP
+      , this.GL.TEXTURE_MAG_FILTER, this.GL.LINEAR);
+
+    // Set up view matrices for each face
+    const captureViews = [
+      lookAt([0, 0, 0], [1, 0, 0], [0, -1, 0]), // Positive X
+      lookAt([0, 0, 0], [-1, 0, 0], [0, -1, 0]), // Negative X
+      lookAt([0, 0, 0], [0, 1, 0], [0, 0, 1]), // Positive Y
+      lookAt([0, 0, 0], [0, -1, 0], [0, 0, -1]), // Negative Y
+      lookAt([0, 0, 0], [0, 0, 1], [0, -1, 0]), // Positive Z
+      lookAt([0, 0, 0], [0, 0, -1], [0, -1, 0]) // Negative Z
+    ];
+
+    // Get the cubemap shader
+    const cubemapShader = this._getCubemapShader();
+    this._pInst.shader(cubemapShader);
+    cubemapShader.setUniform('equirectangularMap', 0);
```

Comment:
This can't be hardcoded to 0. It must have a variable, a sampler2D variable specifically.

Reply:
Actually I had this in mind that in WebGL (and OpenGL), when you set a texture uniform in a shader, you also need to specify which texture unit the texture is bound to.
If you see the next 2 lines:
``` js
this.GL.activeTexture(this.GL.TEXTURE0);
this.GL.bindTexture(this.GL.TEXTURE_2D, input);
```
Here, ` this.GL.activeTexture(this.GL.TEXTURE0)` activates texture unit 0, and `this.GL.bindTexture(this.GL.TEXTURE_2D, input)` binds the texture input to texture unit 0.
But lets just keep it simple and stick to what you have suggested. That seems to work fine.!